stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Events.kt

```diff
@@ -0,0 +1,90 @@
+import io.getstream.chat.android.client.events.DisconnectedEvent
+import io.getstream.chat.android.client.events.NewMessageEvent
+import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
+import io.getstream.chat.android.client.events.UserPresenceChangedEvent
+import io.getstream.chat.android.client.subscribeFor
+import io.getstream.chat.android.client.subscribeForSingle
+import io.getstream.chat.android.client.utils.observable.Disposable
+
+class Events(val client: ChatClient, val channelController: ChannelClient) {
+
+    fun listenSpecificChannelEvents() {
+        // Subscribe for new message events
+        val disposable: Disposable = channelController
+            .subscribeFor<NewMessageEvent> { newMessageEvent ->
+                // to get the message
+                val message = newMessageEvent.message
+            }
+
+        // Dispose when you want to stop receiving events
+        disposable.dispose()
+    }
+
+    fun listenAllChannelEvents() {
+        val disposable: Disposable = channelController
+            .subscribe { event: ChatEvent ->
+                when (event) {
+                    is NewMessageEvent -> {
+                        // to get the message
+                        val message = event.message
+                    }
+                }
+            }
+
+        // Dispose when you want to stop receiving events
+        disposable.dispose()
+    }
+
+    fun listenClientEvents() {
+        // Subscribe for User presence events
+        client.subscribeFor<UserPresenceChangedEvent> { event ->
+            // Handle change
+        }
+
+        // Subscribe for just the first ConnectedEvent
+        client.subscribeForSingle<ConnectedEvent> { event ->
+            // Use event data
+            val unreadCount = event.me.totalUnreadCount
+            val unreadChannels = event.me.unreadChannels
+        }
+    }
+
+    fun listenConnectionEvents() {
+        client.subscribeFor(ConnectedEvent::class, ConnectingEvent::class, DisconnectedEvent::class) { event ->
+            when (event) {
+                is ConnectedEvent -> {
+                    // Socket is connected
+                }
+                is ConnectingEvent -> {
+                    // Socket is connecting
+                }
+                is DisconnectedEvent -> {
+                    // Socket is disconnected
+                }
+            }
+        }
+    }
+
+    fun sendTypingEvents() {
+        // sends a typing.start event if it's been more than 3000 ms since the last event
+        channelController.keystroke().enqueue()
+
+        // sends an event typing.stop to all channel participants
+        channelController.stopTyping().enqueue()
+    }
+
+    fun notificationEvents() {
+        // an example of how listen event when a user is added to a channel
+        channelController.subscribeFor<NotificationAddedToChannelEvent> { notificationEvent ->
+            // Handle event
+        }
+    }
```

Comment:
Took me a moment to figure out that these were on a different page than the other snippets above. What if we matched the structure of the docs with our code here, either by nesting classes or functions?

So in the case of Events
![image](https://user-images.githubusercontent.com/12054216/100992938-0a865480-3555-11eb-93e7-ea0cb88bcf22.png)


Something like

```kotlin
class Events {
    class EventObject {
         // snippet functions
    }
    class ListeningForEvents{
         // snippet functions
    }
}
```

or

```kotlin
class Events {
    fun eventObject() {
         // nested snippet functions
    }
    fun listeningForEvents() {
         // nested snippet functions
    }
}
```

Reply:
I like nested classes solutions. Also, maybe it is worth it to add KDoc with a link to the documentation?

Reply:
Ah good idea, I'd update the `Events` class. Will update for `ClientAndUsers` and `Messages` class in next PR too.